rebasehelper/plugins/build_log_hooks/files.py

```diff
@@ -277,35 +277,52 @@ def _correct_deleted_files(cls, rebase_spec_file, files):
                         i += 1
                         continue
                     split_line = original_line[:]
+                    # Keep track of files which could possibly be renamed but
+                    # not detected by the hook. %doc and %license files are
+                    # the 2 examples of this. If %doc README is renamed to
+                    # README.md, the hook will simply remove it but README.md
+                    # won't be added (it is installed by the directive).
+                    # We want to warn the user about this.
+                    possible_rename = [False for _ in split_line]
                     directives: List[str] = []
                     prepend_macro = None
+                    prepended_directive = None
                     for element in reversed(split_line):
                         if element in cls.FILES_DIRECTIVES:
                             if cls.FILES_DIRECTIVES[element]:
                                 prepend_macro = cls.FILES_DIRECTIVES[element]
+                                prepended_directive = element
                             directives.insert(0, element)
                             split_line.remove(element)
 
                     if prepend_macro:
                         for j, path in enumerate(split_line):
                             if not os.path.isabs(path):
                                 split_line[j] = os.path.join(prepend_macro, subpackage, os.path.basename(path))
+                                possible_rename[j] = True
                     split_line = [MacroHelper.expand(p) for p in split_line]
 
                     j = 0
                     while j < len(split_line) and files:
                         file = split_line[j]
+                        warn_about_rename = possible_rename[j]
                         for deleted_file in reversed(files):
                             if not fnmatch.fnmatch(deleted_file, file):
                                 continue
 
                             original_file = original_line[len(directives) + j]
 
+                            del possible_rename[j]
                             del split_line[j]
                             del original_line[len(directives) + j]
                             files.remove(deleted_file)
                             result['removed'][sec_name].append(original_file)
                             logger.info("Removed %s from '%s' section", original_file, sec_name)
+                            if warn_about_rename:
+                                logger.warning("The installation of %s was done through a %s directive and has now "
+                                               "been removed. The file may have been renamed and rebase-helper cannot "
+                                               "automatically detect it. A common example of this problem is renaming "
+                                               "of README to README.md", original_file, prepended_directive)
```

Comment:
I would change the formulation a bit:

```
The installation of %s was handled by %s directive and the file has now been removed.
The file may have been renamed and rebase-helper cannot automatically detect it.
A common example of this is renaming README to README.md.
```

Reply:
And maybe add another sentence, something like: `It might be necessary to re-add such renamed file to the rebased specfile manually`.